app/browser/api/ledger.js

```diff
@@ -2343,6 +2360,51 @@ const onMediaPublisher = (state, mediaKey, response, duration, revisited) => {
   return state
 }
 
+const getPromotion = () => {
+  let tempClient = client
+  if (!tempClient) {
+    clientprep()
+    tempClient = ledgerClient(null, underscore.extend({roundtrip: roundtrip}, clientOptions), null)
+  }
+
+  tempClient.getPromotion((err, result) => {
+    if (err) {
+      console.error('Error retrieving promotion', err.toString())
+      return
+    }
+
+    appActions.saveLedgerPromotion(result)
+  })
+}
+
+const claimPromotion = (state, promotionId) => {
+  if (!client) {
+    return
+  }
+
+  client.setPromotion(promotionId, (err, result) => {
```

Comment:
getPromotion and setPromotion both use retryTrip. need to be careful that the retry algorithm doesn't trigger whatever we end up using for rate-limiting and protection against DoS (cloudflare?) on the grant server

cc @evq